src/components/Modals/UploadDocumentModal.jsx

```diff
@@ -64,46 +81,78 @@ const UploadDocumentModal = ({ showModal, setShowModal }) => {
     setShowModal(false);
   };
 
-  // Event handler for form/document submission to Pod
-  const handleDocUpload = async (e) => {
-    e.preventDefault();
-    setProcessing(true);
+  const cleanup = () => {
+    setShowConfirmationModal(false);
+    setConfirmationModalType('add');
+    setProcessing(false);
+    clearInputFields();
+  };
 
+  const handleDocUpload = async (uploadType) => {
     const fileDesc = {
       name: file.name,
       type: docType,
       date: expireDate,
       description: docDescription
     };
 
-    try {
-      await addDocument(fileDesc, file);
-      addNotification('success', `File uploaded to Solid.`);
-    } catch (error) {
-      const confirmationMessage =
-        'A file of this name and type already exists on the pod. Would you like to replace it?';
-
-      switch (error.message) {
-        case 'File already exists':
-          if (window.confirm(confirmationMessage)) {
-            await replaceDocument(fileDesc, file);
-            addNotification('success', `File updated on Solid.`);
+    switch (uploadType) {
+      case 'add':
+        try {
+          await addDocument(fileDesc, file);
+          addNotification('success', `File uploaded to Solid.`);
+          cleanup();
+        } catch (error) {
+          switch (error.message) {
+            case 'File already exists':
+              // The confirmation modal will prompt the user to replace the file or not
+              setConfirmationModalType('replace');
+              setShowConfirmationModal(true);
+              break;
+            default:
+              addNotification('error', `File failed to upload. Reason: ${error.message}`);
           }
-          break;
-        default:
+        }
+        break;
+      case 'replace':
+        try {
+          await replaceDocument(fileDesc, file);
+          addNotification('success', `File updated on Solid.`);
+          cleanup();
+        } catch (error) {
           addNotification('error', `File failed to upload. Reason: ${error.message}`);
-      }
-    } finally {
-      setProcessing(false);
-      clearInputFields();
+        }
+        break;
+      default:
+        throw new Error('handleDocUpload: Invalid uploadType');
     }
   };
 
+  const handleUploadCancelled = () => {
+    setShowConfirmationModal(false);
+    setConfirmationModalType('add');
+    setProcessing(false);
+  };
+
+  // Event handler for form/document submission to Pod
+  const onFormSubmit = async (e) => {
+    e.preventDefault();
+    setProcessing(true);
+    handleDocUpload('add');
+  };
+
   return (
     <Dialog open={showModal} aria-labelledby="upload-document-dialog" onClose={clearInputFields}>
+      <ConfirmationModal
+        {...confirmationModalContentVariant[confirmationModalType]}
```

Comment:
While I do like that we're able to make the confirmation modal dynamic for the  `<UploadDocumentsModal>`, was wondering if we should just make it into its own specialized component.

Reply:
I agree that would probably make things more clear. I'll give it a try!

Reply:
Okay, I made a specialized component, like you suggested, and I refactored some of the code for clarity...